### PR TITLE
Add support for temporary URLs when using S3 as storage

### PR DESCRIPTION
**Pull request by PimButton**
This PR adds support for using S3 as a non-public storage engine when also using responsive images. Currently the combination between S3 and responsive images only works when the S3 repo bucket is public. This is not always possible so with this PR I hope to add support for  this combination. 

This is my first contribution to this project and I have not written any tests yet. I know this is required before this PR can even be considered so can anyone point me in the right direction as to how to write a test for this new functionality? Many thanks!

### CODE REVIEW

**Comment by PimButton** [comment]
Ping @freekmurze for notice :)

**Comment by freekmurze** [comment]
Closing due to inactivity

**Comment by PimButton** [comment]
Whoops, I did not get a notification of your reply! Will pick this up asap :)

File: src/UrlGenerator/S3UrlGenerator.php [hunk 1]
```diff
@@ -71,4 +71,21 @@ public function getResponsiveImagesDirectoryUrl(): string
...
+    /**
+     * Get the temporary url for a responsive media item.
+     *
+     * @param string $filename
+     * @param \DateTimeInterface $expiration
+     * @param array $options
+     *
+     * @return string
+     */
+    public function getTemporaryResponsiveImageUrl(string $filename, DateTimeInterface $expiration, array $options = []): string
```
[the remark this answers is not in the document]
Comment (PimButton) [reply src/UrlGenerator/S3UrlGenerator.php]: I used the same name as the default `getTemporaryUrl` function, but `expiresAt` makes more sense indeed. Have changed it in my function now

[the remark this answers is not in the document]
Comment (PimButton) [reply src/UrlGenerator/S3UrlGenerator.php]: I have added an exception when using the Local Url Generator

**Comment by PimButton** [comment]
@freekmurze did you have time to take a look at this already?

**Comment by PimButton** [comment]
@freekmurze just another ping ;)

**Comment by freekmurze** [comment]
Thank you for your work, but I've decided not to add it to the package. Main reason is because we don't have a need for this ourselves and want to keep the package as simple as it can be.

**Comment by JhumanJ** [comment]
@freekmurze Any chance you would consider integrating this as of today ? Would be super helpful

**Pull request closed without merging**